### PR TITLE
fix: bats test

### PR DESCRIPTION
**Pull request by atanmarko**
# Description

Add bats dependencies as submodules of the project.
Updated `_common_setup`.

## PR Checklist:

- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added or updated tests that comprehensively prove my change is effective or that my feature works

### CODE REVIEW

File: Makefile.e2e.toml [hunk 1]
```diff
@@ -23,5 +23,5 @@ category = "E2E - PREPARE"
 cwd = "tests/kurtosis"
 script = '''
 kurtosis clean --all
-kurtosis run --enclave cdk --args-file .github/tests/fork12-pessimistic.yml --image-download always . '{"args": {"agglayer_prover_sp1_key": "\'$AGGLAYER_PROVER_SP1_KEY'"}}'
+kurtosis run --enclave cdk --args-file .github/tests/fork12-pessimistic.yml --image-download always . '{"args": {"agglayer_prover_sp1_key": "'$AGGLAYER_PROVER_SP1_KEY'"}}'
```
Comment (0xaatif) [inline Makefile.e2e.toml:26]: very nice :)

**Review by 0xaatif — COMMENTED** [review]
Love this, but I think submodules are a bit of a pain - can you think of another way? Is there a reason `homebrew` isn't the best solution?

**Comment by atanmarko** [comment]
> Is there a reason homebrew isn't the best solution?  

https://github.com/ztombol/bats-docs?tab=readme-ov-file#installation

Documentation says it is only for `macOS` users. Many people have Linux. Even if you can setup `brew` on your system, I am not happy to introduce another package manager to my system just to be able to install two supporting libraries.

>  Do they not ship bats-assert and bats-core?

Homebrew, npm and source code.

File: .gitmodules
Comment (Freyskeyd) [inline .gitmodules:1]: I would prefer relying on npm version of those libraries: 

- https://www.npmjs.com/package/bats-assert
- https://www.npmjs.com/package/bats-support

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
 target
 tests/kurtosis
+.idea
```
Comment (Freyskeyd) [inline .gitignore:3]: I think that could be place in a global `.gitignore` on your machine ^^

**Pull request closed without merging**